gpt4all-chat/modellist.h

```diff
@@ -488,7 +498,7 @@ private Q_SLOTS:
 
 protected:
     explicit ModelList();
-    ~ModelList() {}
+    ~ModelList() { for (auto *model: m_models) { delete model; } }
```

Comment:
This will never get called as it is a singleton

Reply:
At some point, the user will close GPT4All. Then main() will return, and global destructors will run. This is a global destructor, and adding this line significantly shortens the LeakSanitizer report for GPT4All.